backend/project404_t8/API/viewsets.py

```diff
@@ -266,21 +240,23 @@ def profileView(request, username):
     profile_posts = Post.objects.filter(author=request.user.id)
     return render(request, 'profile/profile.html', {'user':user, "posts":profile_posts})
 
-    
-
-    return render(request, 'post/post.html', {"post":post, "imageExists":imageExists})
-
-def IndexView(request):
-
-    # This is our post object with the given ID
-    post = Post.objects.all() #pk is primary key
-
-    # We could check to see if the user has permission to view this post in here
-    # Based on the privacy setting etc.
-    # imageExists = False
-    # if post.image_link != "":
-    #     imageExists = True
-
-    # filter by yourself,friends, friends of friends
-    return render(request, 'homepage/home.html', {"post":post})
-    
+def postsListView(request):
+    try:
+        uname = request.user
+        uid = uname.id
+        post = Post.objects.raw(' \
+        WITH posts AS (SELECT id FROM API_post WHERE author_id in  \
+        (SELECT f2.friend_a_id AS fofid \
+            FROM API_friendship f \
+            JOIN API_friendship f2 ON f.friend_a_id = f2.friend_b_id \
+            WHERE fofid NOT IN (SELECT friend_a_ID FROM API_friendship  \
+            WHERE friend_b_id = %s) AND f.friend_b_id = %s AND fofid != %s) AND privacy_setting = 4 \
+        UNION \
+            SELECT id FROM API_post WHERE (author_id in  \
+            (WITH friends(fid) AS (SELECT friend_b_id FROM API_friendship WHERE friend_a_id=%s) \
+            SELECT * FROM friends WHERE fid != %s GROUP BY fid)  \
+            AND privacy_setting = 3 OR privacy_setting = 4) OR author_id = %s OR  privacy_setting = 6) \
+            SELECT * FROM API_post WHERE id in posts', [int(uid)]*6)
+    except:
+        post = Post.objects.all()
```

Comment:
is this in the case where there's some sort of error with the sqlite query? just out of curiosity

Reply:
it's for the logout-anonymous user error. this makes it safe to logout and still request this list but get nothing back when you lick logout.